av_metrics_tool/src/main.rs

```diff
@@ -116,124 +145,126 @@ enum AudioContainer {
-        );
-    }
+    let mut results = MetricsResults::default();
 
     if metric.is_none() || metric == Some("psnr") {
-        let psnr = Psnr::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("psnr", hashmap! {"result" => psnr});
-        }
+        results.psnr = Psnr::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
     if metric.is_none() || metric == Some("apsnr") {
-        let apsnr = APsnr::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("apsnr", hashmap! {"result" => apsnr});
-        }
+        results.apsnr = APsnr::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
     if metric.is_none() || metric == Some("psnrhvs") {
-        let psnrhvs = PsnrHvs::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("psnrhvs", hashmap! {"result" => psnrhvs});
-        }
+        results.psnr_hvs = PsnrHvs::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
     if metric.is_none() || metric == Some("ssim") {
-        let ssim = Ssim::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("ssim", hashmap! {"result" => ssim});
-        }
+        results.ssim = Ssim::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
     if metric.is_none() || metric == Some("msssim") {
-        let msssim = MsSsim::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("msssim", hashmap! {"result" => msssim});
-        }
+        results.msssim = MsSsim::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
     if metric.is_none() || metric == Some("ciede2000") {
-        let ciede2000 = Ciede2000::run(
-            input1.as_ref(),
-            container1,
-            input2.as_ref(),
-            container2,
-            serialize,
-        );
-        if serialize {
-            results.insert("ciede2000", hashmap! {"result" => ciede2000});
-        }
+        results.ciede2000 =
+            Ciede2000::run(input1.as_ref(), container1, input2.as_ref(), container2);
     }
 
-    if serialize {
-        print!("{}", serde_json::to_string(&results).unwrap());
+    match serialize {
+        OutputType::JSON => println!("{}", serde_json::to_string(&results).unwrap()),
```

Comment:
Would be better to write to file.

Reply:
av-metrics-tool being a command-line tool, the output to stdout can easily be redirected to a file if needed and can also be piped to another command. 
Making it only write to a file would restrict use cases.
As of writing to file, wouldn't it be worth opening another issue for this feature ?

Reply:
Would be a good idea to split that indeed. Feel free to open another pr to put the csv output to a file and rebase this on top.